streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableReduce.java

```diff
@@ -73,7 +73,17 @@ public void process(final K key, final Change<V> value) {
             final V oldAgg = store.get(key);
             V newAgg = oldAgg;
 
-            // first try to add the new value
+            // first remove the old value
+            if (value.oldValue != null) {
+                if (newAgg == null) {
+                    throw new StreamsException(
+                        "Previous aggregate for key=[" + key + "] should not be null at this point. Has the topology changed?"
```

Comment:
nit: Is topology changing the only possible root cause of this issue? I felt `Has the topology changed?` may be a bit too specific for guiding users to trouble shoot.

Reply:
I think that a topology change is the only mechanism we currently have of violating this invariant.

Obviously, I can't be sure, though. We can omit the second sentence, or we can hedge it: "This might be due to an illegal change in the topology."

WDYT?